### PR TITLE
Refactor all instances of frequent to bookmark

### LINKED ISSUE #186: Rename all instances of frequent to bookmark
Rename all classes, methods and variables from frequent to bookmark.

### PR DESCRIPTION
**Pull request by siddarth2824**
Resolves #186.
Refactored all instances of frequent to bookmark.

### CODE REVIEW

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/commons/core/Messages.java [hunk 1]
```diff
@@ -10,9 +10,9 @@
     public static final String MESSAGE_INVALID_TAB_FORMAT = "'%s' command can only be used in the following tabs: %s";
     public static final String MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX =
             "The transaction index provided is invalid.";
-    public static final String MESSAGE_INVALID_FREQUENT_EXPENSE_DISPLAYED_INDEX =
+    public static final String MESSAGE_INVALID_BOOKMARK_EXPENSE_DISPLAYED_INDEX =
             "The frequent expense index provided is invalid";
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/commons/core/Messages.java:14]:
```suggestion
            "The bookmark expense index provided is invalid";
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/commons/core/Messages.java [hunk 2]
```diff
@@ -10,9 +10,9 @@
...
-    public static final String MESSAGE_INVALID_FREQUENT_INCOME_DISPLAYED_INDEX =
+    public static final String MESSAGE_INVALID_BOOKMARK_INCOME_DISPLAYED_INDEX =
             "The frequent income index provided is invalid";
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/commons/core/Messages.java:16]:
```suggestion
            "The bookmark income index provided is invalid";
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/ConvertBookmarkIncomeCommand.java [hunk 3]
```diff
@@ -11,51 +11,51 @@
...
+    public static final String MESSAGE_CONVERT_BOOKMARK_INCOME_SUCCESS = "Bookmark income has been converted "
             + "and successfully added to finance tracker: %1$s";
 
     private final Index targetIndex;
     private final Date date;
 
     /**
-     * @param targetIndex Index of the frequent income in the filtered frequent income list to convert.
-     * @param convertDate Date of converting a frequent income to an income and adding it to the finance tracker.
+     * @param targetIndex Index of the bookmark income in the filtered frequent income list to convert.
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/ConvertBookmarkIncomeCommand.java:38]:
```suggestion
     * @param targetIndex Index of the bookmark income in the filtered bookmark income list to convert.
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/DeleteBookmarkIncomeCommand.java [hunk 4]
```diff
@@ -10,46 +10,46 @@
...
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the frequent income identified by the index number used in the displayed "
-            + "frequent income list. \n"
+            + ": Deletes the bookmark income identified by the index number used in the displayed "
+            + "bookmark income list. \n"
             + "Parameters: INDEX (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1 ";
 
-    public static final String MESSAGE_DELETE_FREQUENT_INCOME_SUCCESS = "Deleted Frequent Income: %1$s";
+    public static final String MESSAGE_DELETE_BOOKMARK_INCOME_SUCCESS = "Deleted Frequent Income: %1$s";
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/DeleteBookmarkIncomeCommand.java:27]:
```suggestion
    public static final String MESSAGE_DELETE_BOOKMARK_INCOME_SUCCESS = "Deleted Bookmark Income: %1$s";
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/EditBookmarkIncomeCommand.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            + PREFIX_AMOUNT + "5 ";
+
+    public static final String MESSAGE_EDIT_BOOKMARK_INCOME_SUCCESS = "Edited Bookmark Income: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index targetIndex;
+    private final EditBookmarkTransactionDescriptor editBookmarkIncomeDescriptor;
+
+    /**
+     * @param targetIndex Index of the frequent income in the filtered frequent income list to edit.
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/EditBookmarkIncomeCommand.java:48]:
```suggestion
     * @param targetIndex Index of the bookmark income in the filtered bookmark income list to edit.
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/EditBookmarkIncomeCommand.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+     * @param editBookmarkIncomeDescriptor Details to edit the frequent income with.
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmark/EditBookmarkIncomeCommand.java:49]:
```suggestion
     * @param editBookmarkIncomeDescriptor Details to edit the bookmark income with.
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/BookmarkIncome.java [hunk 7]
```diff
@@ -33,12 +33,12 @@ public boolean equals(Object other) {
             return true;
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/BookmarkIncome.java:33]: The javadoc for this method should be changed from frequent incomes to bookmark incomes (line 27 and 28).

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/exceptions/DuplicateBookmarkTransactionException.java [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
+package ay2021s1_cs2103_w16_3.finesse.model.bookmark.exceptions;
 
 /**
  * Signals that the following operation will result in a duplicate frequent transaction.
  */
-public class DuplicateFrequentTransactionException extends RuntimeException {
-    public DuplicateFrequentTransactionException(String message) {
+public class DuplicateBookmarkTransactionException extends RuntimeException {
+    public DuplicateBookmarkTransactionException(String message) {
         super("Operation would result in duplicate frequent " + message);
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/exceptions/DuplicateBookmarkTransactionException.java:8]:
```suggestion
        super("Operation would result in duplicate bookmark " + message);
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/exceptions/DuplicateBookmarkTransactionException.java [hunk 9]
```diff
@@ -1,10 +1,10 @@
-package ay2021s1_cs2103_w16_3.finesse.model.frequent.exceptions;
+package ay2021s1_cs2103_w16_3.finesse.model.bookmark.exceptions;
 
 /**
  * Signals that the following operation will result in a duplicate frequent transaction.
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/exceptions/DuplicateBookmarkTransactionException.java:4]:
```suggestion
 * Signals that the following operation will result in a duplicate bookmark transaction.
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/tabs/ExpenseTabPane.java [hunk 10]
```diff
@@ -18,10 +18,10 @@
      * and bookmarked expenses.
      *
      * @param expenseList The list of expenses to be displayed.
-     * @param frequentExpenseList The list of frequent expenses to be displayed.
+     * @param bookmarkExpenseList The list of frequent expenses to be displayed.
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/tabs/ExpenseTabPane.java:21]:
```suggestion
     * @param bookmarkExpenseList The list of bookmark expenses to be displayed.
```

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/tabs/IncomeTabPane.java [hunk 11]
```diff
@@ -18,10 +18,10 @@
      * and bookmarked incomes.
      *
      * @param incomeList The list of incomes to be displayed.
-     * @param frequentIncomeList The list of frequent incomes to be displayed.
+     * @param bookmarkIncomeList The list of frequent incomes to be displayed.
```
Comment (zhaojj2209) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/tabs/IncomeTabPane.java:21]:
```suggestion
     * @param bookmarkIncomeList The list of bookmark incomes to be displayed.
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java [hunk 12]
```diff
@@ -144,15 +144,15 @@ public static void assertCommandFailure(Command command, Model actualModel, Stri
         // only do so by copying its components.
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java:144]: Line 139 should be changed from frequent expense list to bookmark expense list.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkExpenseCommandTest.java [hunk 13]
```diff
@@ -14,98 +14,98 @@
...
     }
 
     @Test
     public void execute_invalidIndexFilteredList_throwsCommandException() {
-        showFrequentExpenseAtIndex(model, INDEX_FIRST);
+        showBookmarkExpenseAtIndex(model, INDEX_FIRST);
 
         Index outOfBoundIndex = INDEX_SECOND;
         Date dateOfConvertedExpense = new Date(VALID_DATE_SPOTIFY_SUBSCRIPTION);
         // ensures that outOfBoundIndex is still in bounds of frequent expense list
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkExpenseCommandTest.java:86]:
```suggestion
        // ensures that outOfBoundIndex is still in bounds of frequent bookmark list
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkExpenseCommandTest.java [hunk 14]
```diff
@@ -14,98 +14,98 @@
...
     }
 
     @Test
     public void equals() {
         Date dateOfConvertedExpense = new Date(VALID_DATE_SPOTIFY_SUBSCRIPTION);
-        ConvertFrequentExpenseCommand convertFirstFrequentExpenseCommand =
-                new ConvertFrequentExpenseCommand(INDEX_FIRST, dateOfConvertedExpense);
-        ConvertFrequentExpenseCommand convertSecondFrequentExpenseCommand =
-                new ConvertFrequentExpenseCommand(INDEX_SECOND, dateOfConvertedExpense);
+        ConvertBookmarkExpenseCommand convertFirstFrequentExpenseCommand =
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkExpenseCommandTest.java:98]:
```suggestion
        ConvertBookmarkExpenseCommand convertFirstBookmarkExpenseCommand =
```
The corresponding references to this name should be changed as well.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkExpenseCommandTest.java [hunk 15]
```diff
@@ -14,98 +14,98 @@
...
+                new ConvertBookmarkExpenseCommand(INDEX_FIRST, dateOfConvertedExpense);
+        ConvertBookmarkExpenseCommand convertSecondFrequentExpenseCommand =
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkExpenseCommandTest.java:100]:
```suggestion
        ConvertBookmarkExpenseCommand convertSecondBookmarkExpenseCommand =
```
Similarly for this one

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkIncomeCommandTest.java [hunk 16]
```diff
@@ -14,97 +14,97 @@
...
     }
 
     @Test
     public void execute_invalidIndexFilteredList_throwsCommandException() {
-        showFrequentIncomeAtIndex(model, INDEX_FIRST);
+        showBookmarkIncomeAtIndex(model, INDEX_FIRST);
 
         Index outOfBoundIndex = INDEX_SECOND;
         Date dateOfConvertedIncome = new Date(VALID_DATE_SPOTIFY_SUBSCRIPTION);
         // ensures that outOfBoundIndex is still in bounds of frequent income list
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/ConvertBookmarkIncomeCommandTest.java:85]: The edits for this file mirror that of `ConvertBookmarkExpenseCommandTest.java`.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/DeleteBookmarkExpenseCommandTest.java [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+
+        assertCommandSuccess(deleteBookmarkExpenseCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showBookmarkExpenseAtIndex(model, INDEX_FIRST);
+
+        Index outOfBoundIndex = INDEX_SECOND;
+        // ensures that outOfBoundIndex is still in bounds of frequent expense list
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/DeleteBookmarkExpenseCommandTest.java:73]:
```suggestion
        // ensures that outOfBoundIndex is still in bounds of bookmark expense list
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/DeleteBookmarkIncomeCommandTest.java [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+
+        assertCommandSuccess(deleteBookmarkIncomeCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showBookmarkIncomeAtIndex(model, INDEX_FIRST);
+
+        Index outOfBoundIndex = INDEX_SECOND;
+        // ensures that outOfBoundIndex is still in bounds of frequent income list
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/DeleteBookmarkIncomeCommandTest.java:74]:
```suggestion
        // ensures that outOfBoundIndex is still in bounds of bookmark income list
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/EditBookmarkExpenseCommandTest.java [hunk 19]
```diff
@@ -90,29 +90,29 @@ public void execute_invalidFrequentExpenseIndexUnfilteredList_failure() {
      * but smaller than size of finance tracker
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/EditBookmarkExpenseCommandTest.java:90]: Line 89 should be changed from frequent expense list to bookmark expense list.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/EditBookmarkExpenseCommandTest.java [hunk 20]
```diff
@@ -90,29 +90,29 @@ public void execute_invalidFrequentExpenseIndexUnfilteredList_failure() {
...
      */
     @Test
-    public void execute_invalidFrequentExpenseIndexFilteredList_failure() {
-        showFrequentExpenseAtIndex(model, INDEX_FIRST);
+    public void execute_invalidBookmarkExpenseIndexFilteredList_failure() {
+        showBookmarkExpenseAtIndex(model, INDEX_FIRST);
         Index outOfBoundIndex = INDEX_SECOND;
 
         // ensures that outOfBoundIndex is still in bounds of frequent expense list
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/EditBookmarkExpenseCommandTest.java:97]:
```suggestion
        // ensures that outOfBoundIndex is still in bounds of bookmark expense list
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/EditBookmarkIncomeCommandTest.java [hunk 21]
```diff
@@ -18,101 +18,101 @@
...
-                new EditFrequentIncomeCommand(outOfBoundIndex, descriptor);
+        EditBookmarkIncomeCommand editBookmarkIncomeCommand =
+                new EditBookmarkIncomeCommand(outOfBoundIndex, descriptor);
 
-        assertCommandFailure(editFrequentIncomeCommand, model, MESSAGE_INVALID_FREQUENT_INCOME_DISPLAYED_INDEX);
+        assertCommandFailure(editBookmarkIncomeCommand, model, MESSAGE_INVALID_BOOKMARK_INCOME_DISPLAYED_INDEX);
     }
 
     /**
      * Edit filtered frequent income list where index is larger than size of filtered frequent income list,
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/bookmarkcommands/EditBookmarkIncomeCommandTest.java:89]: The edits in this file mirror that of `EditBookmarkExpenseCommandTest.java`.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/BookmarkExpenseTest.java [hunk 22]
```diff
@@ -32,18 +32,18 @@ public void equals() {
         assertFalse(PHONE_BILL.equals(SPOTIFY_SUBSCRIPTION));
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/BookmarkExpenseTest.java:32]: Line 31 should be changed from frequent expense to bookmark expense.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/BookmarkIncomeTest.java [hunk 23]
```diff
@@ -33,18 +33,18 @@ public void equals() {
         assertFalse(PART_TIME.equals(INVESTING));
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/model/bookmark/BookmarkIncomeTest.java:33]: Similar change to BookmarkExpenseTest.java.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TypicalTransactions.java [hunk 24]
```diff
@@ -60,20 +60,20 @@
             .withDate("22/09/2020").withAmount("1000").buildIncome();
 
     // Frequent Expenses
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TypicalTransactions.java:62]:
```suggestion
    // Bookmark Expenses
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TypicalTransactions.java [hunk 25]
```diff
@@ -60,20 +60,20 @@
...
+    public static final BookmarkExpense PHONE_BILL = new BookmarkTransactionBuilder().withTitle("Phone Bill")
+            .withAmount("60").buildBookmarkExpense();
+    public static final BookmarkExpense SPOTIFY_SUBSCRIPTION = new BookmarkTransactionBuilder()
+            .withTitle("Spotify Subscription").withAmount("9.90").buildBookmarkExpense();
+    public static final BookmarkExpense NETFLIX_SUBSCRIPTION = new BookmarkTransactionBuilder()
+            .withTitle("Netflix Subscription").withAmount("20").buildBookmarkExpense();
+    public static final BookmarkExpense TIMES_MAGAZINE_SUBSCRIPTION = new BookmarkTransactionBuilder()
+            .withTitle("Times Magazine Subscription").withAmount("12").buildBookmarkExpense();
 
     // Frequent Incomes
```
Comment (zhaojj2209) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TypicalTransactions.java:72]:
```suggestion
    // Bookmark Incomes
```

**Review by zhaojj2209 — CHANGES_REQUESTED** [review]
Mostly javadoc changes!

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

File: docs/diagrams/ConvertBookmarkExpenseSequenceDiagram.puml [hunk 26]
```diff
@@ -0,0 +1,68 @@
...
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":FinanceTrackerParser" as FinanceTrackerParser LOGIC_COLOR
+participant ":ConvertBookmarkExpenseCommand" as ConvertBookmarkExpenseCommand LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":ModelManager" as ModelManager MODEL_COLOR
+end box
+
+box FrequentExpense UI_COLOR_T1
```
Comment (ianyong) [inline docs/diagrams/ConvertBookmarkExpenseSequenceDiagram.puml:14]:
```suggestion
box BookmarkExpense UI_COLOR_T1
```

File: docs/diagrams/ConvertBookmarkExpenseSequenceDiagram.puml [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+participant ":BookmarkExpense" as BookmarkExpense UI_COLOR
+end box
+
+[->LogicManager: execute("convert-frequent-expense 1 d/20/10/2020")
```
Comment (ianyong) [inline docs/diagrams/ConvertBookmarkExpenseSequenceDiagram.puml:18]:
```suggestion
[->LogicManager: execute("convert-bookmark-expense 1 d/20/10/2020")
```

**Review by ianyong — CHANGES_REQUESTED** [review]
For the `UserGuide.md`, the links at lines 65 and lines 72 need to be updated as well.

File: docs/DeveloperGuide.md [hunk 28]
```diff
@@ -246,7 +246,7 @@ conversion.
 Given below is the proposed UML sequence diagram and explanation of an example usage scenario for
 convert-frequent-expense
 
-![Sequence Diagram of the Convert Frequent Expense To Expense Feature](images/ConvertFrequentExpenseSequenceDiagram.png)
+![Sequence Diagram of the Convert Frequent Expense To Expense Feature](images/ConvertBookmarkExpenseSequenceDiagram.png)
```
Comment (ianyong) [inline docs/DeveloperGuide.md:249]:
```suggestion
![Sequence Diagram of the Convert Bookmark Expense To Expense Feature](images/ConvertBookmarkExpenseSequenceDiagram.png)
```

**Review by ianyong — APPROVED** [review]
LGTM! Note that the developer guide has yet to be fully refactored. This will be done when we update the developer guide at a later date.

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Pull request merged**